### PR TITLE
Add User schema and paths

### PR DESCRIPTION
**Pull request by vpchung**
With this PR:

- adding `User` schema and other related schemas (#47)
- adding paths and operations for `User`
- updating `openapi/openapi.yaml` to include the changes

For this schema, I am using [Google Workspace user API](https://developers.google.com/admin-sdk/directory/reference/rest/v1/users) as reference.  I think it has some properties that we could potentially utilize in the future.

### CODE REVIEW

**Comment by vpchung** [comment]
Not fully fleshed out, so converting to draft.

**Comment by vpchung** [comment]
@tschaffter not ready yet, sorry!

There's one thing I need to fix for the User* schemas, regarding `password`.

File: openapi/components/schemas/PageOfUsers.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+description: A page of Users
+allOf:
+  - $ref: ResponsePageMetadata.yaml
+  - type: object
+    properties:
+      users:
+        description: An array of Users
+        type: array
+        items:
+          $ref: UserGetResponse.yaml
```
Comment (tschaffter) [inline openapi/components/schemas/PageOfUsers.yaml:11]: This is less than ideal. Let's target to return a `User` object (does not include properties like `password` and `salt`). Out of curiosity, have you seen `UserGetResponse` used in another project (Google?).

File: openapi/components/schemas/UserGetResponse.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+type: object
+description: A user of the portal
```
Comment (tschaffter) [inline openapi/components/schemas/UserGetResponse.yaml:2]: Replace by "A user of the service"? One can make a query without using the portal.

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -28,8 +28,8 @@ tags:
     description: Operations about persons
   - name: Tag
     description: Operations about tags
-  # - name: User
-  #   description: Operations about users
+  - name: User
+    description: Operations about users
 servers:
```
Comment (tschaffter) [inline openapi/openapi.yaml:33]: Tom told me that he had issues working on a client generated from an `openapi.json` that has the property `servers` generated as we do it here (I forgot the specifics). In the NLP Sandbox Data Node I went back to

```
servers:
  - http://...
  - https://..
```

Let's keep this format here unless you meet an issue when working with the Python client.

**Review by tschaffter — CHANGES_REQUESTED** [review]
Looks good! I have a few comments and requests.

File: openapi/components/schemas/PageOfUsers.yaml (hunk 1 above)
Comment (vpchung) [reply openapi/components/schemas/PageOfUsers.yaml:11]: I have not. I created `UserGetResponse` solely for the purpose of returning a payload that included all the `User` properties except for `password`.

I'll go ahead and remove the `password` property for now, until I look further into authorization/authentication!

File: openapi/components/schemas/UserGetResponse.yaml (hunk 2 above)
Comment (vpchung) [reply openapi/components/schemas/UserGetResponse.yaml:2]: Makes sense!

File: openapi/openapi.yaml (hunk 3 above)
Comment (vpchung) [reply openapi/openapi.yaml:33]: Good to know!

To be honest, this was something I was trying out as I was reading through the OpenAPI docs and forgot to change back.  I'm not sure if it actually serves any benefit over the approach used in NLP Sandbox.

Comment (vpchung) [reply openapi/openapi.yaml:33]: Good to know!

To be honest, this was something I was trying out as I was reading through the OpenAPI docs and forgot to change back. I'm not sure if it actually serves any benefit over the approach used in NLP Sandbox.

**Comment by tschaffter** [comment]
Looks great!

**Pull request merged**